### ISSUE TITLE
popup, modal: improve focus behaviour

### ISSUE THREAD

**Issue by aristov** [bem/bem-components#1673]
Here are the recommendations from the 2nd and 3rd points of [Understanding WCAG: Focus order](http://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html#navigation-mechanisms-focus-order-examples-head):

for modeless dialogs:
- [x] ~~When the trigger button is activated, a dialog opens~~
- [ ] The interactive elements in the dialog are inserted in the focus order immediately after the button
- [ ] When the dialog is open, the focus order goes from the button to the elements of the dialog, then to the interactive element following the button
- [x] ~~When the dialog is closed, the focus order goes from the button to the following element~~

for modal dialogs:
- [x] ~~When the trigger button is activated, a dialog opens~~
- [ ] and focus is set to the first interactive element in the dialog
- [ ] As long as the dialog is open, focus is limited to the elements of the dialog
- [ ] When the dialog is dismissed, focus returns to the button or the element following the button

Current implementations satisfy the crossed out items. We should implement the rest.

**Comment by aristov** [comment]
Current `popup` and `modal` implementations are too common cases to manage focus by themselves. E.g. `popup` may be used to create some kind of "tooltip" with text but without focusable elements inside.
All the features described above should be implemented by the dialog box, which must contain at least one interactive component. So I close this issue, as we have no implementation of a dialog box yet.
